src/js/meteoData.js

```diff
@@ -0,0 +1,169 @@
+        this.temp = temp;
+        this.temp_min = temp_min;
+        this.temp_max = temp_max;
+        this.weather = weather;
+        this.description = description;
+        this.icon = icon;
+        this.wind_speed = wind_speed;
+        this.pressure = pressure;
+        this.humidity = humidity;
+    }
+}
+
+ async function getMeteoData(cityName){
+    const currentWeatherObject = await toDataObject(`http://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&appid=70e6060b22d5d79f05a17ef7fadfdbda`); 
+    const forecastWeatherObject = await toDataObject(`http://api.openweathermap.org/data/2.5/forecast?q=${cityName}&units=metric&appid=70e6060b22d5d79f05a17ef7fadfdbda`);
+
+    const resultArray = [];
+    const dateTime  = todayDate();
+
+    const meteo = new meteoObject(dateTime, Math.floor(currentWeatherObject.main.temp), Math.floor(currentWeatherObject.main.temp_min), 
+      Math.floor(currentWeatherObject.main.temp_max), currentWeatherObject.weather[0].main, currentWeatherObject.weather[0].description, 
+      currentWeatherObject.weather[0].icon, currentWeatherObject.wind.speed, currentWeatherObject.main.pressure, currentWeatherObject.main.humidity);
+
+    resultArray.push(meteo);
+    const dividedArray = divideArray(forecastWeatherObject.list);
+    for(let i = 1; i < dividedArray.length; i++){
+        resultArray.push(transformToObject(dividedArray[i]));
+    }
+
+    return resultArray;
+  }
+
+  //funkcja pobiera dane z api i zamienia w obiekt 
+
+  async function toDataObject(fetchLink){
+    const fetchedData = await fetch(fetchLink); //pobieranie danych z API
+    const stringData = await fetchedData.text(); //konwersja pobranych danych w obiekt zapisany w postaci stringa
+    const dataObject = JSON.parse(stringData); //konwersja obiektu zapisanego w postaci stringa w prawdziwy obiekt
+  
+    return dataObject;
+  }
+
+  //todayDate zwraca obecna date
+
+  function todayDate(){ 
+    const today = new Date();
+    const date = today.getFullYear() + '-' + (today.getMonth()+1) + '-' + today.getDate();
+    const time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
+    const dateTime = date + ' ' + time;
+    return dateTime;
+  }
+  
+  /* 
+    Funkcja divideArray służy mi do tego aby podzielić 40 elementową tablice zawierającą prognozy pogody co 3 godziny na 6 elementową, w której każdy element
+    reprezentuje zbiór tablic prognoz odnoszących się do 1 dnia, ułatwia to póżniejsza przemiane tych tablic na dane odnośnie prognoz na dany dzień
+  */
+  function divideArray(meteoArr){
+    let dayArray = [];
+    const dividedArray = [];
+  
+    let data1 = new Date(meteoArr[0].dt_txt);
+    data1 = data1.getDay();
+    dayArray.push(meteoArr[0])
+
+    for(let i = 1; i < meteoArr.length; i++){//for w tym miejscu ponieważ zaczynam od drugiego elementu, a z tego co kojarzę wszystkie iteratory obejmują wszystkie elementy
+        let data2 = new Date(meteoArr[i].dt_txt);
```

Comment:
czemu od drugiego?

Reply:
Faktycznie mogę to zrobić forEachem, nie wiem czemu wcześniej nie pomyślałem o tym.

Reply:
czemu nie mapem?